frontend/taipy-gui/src/components/Taipy/FileSelector.tsx

```diff
@@ -65,12 +68,14 @@ const FileSelector = (props: FileSelectorProps) => {
         defaultLabel = "",
         updateVarName = "",
         multiple = false,
+        selectFolder = false,
         extensions = ".csv,.xlsx",
         dropMessage = "Drop here to Upload",
         label,
         notify = true,
         withBorder = true,
     } = props;
+    const directoryProps = selectFolder ? {webkitdirectory: "", directory: "", mozdirectory: "", nwdirectory: ""}:undefined;
```

Comment:
useMemo

Reply:
I don't get your comment.

Reply:
I meant to define directoryProps via a useMemo in the main component but as the resulting object is not considered for the render calculation it doesn't matter